Setup manager feature

PartySensei opened the pull request:
- Introduces Setup Manager as a stand alone feature;
- Small amendments in ArmoryQt.py for the setup manager to work as intended;
- All UI elements used by Setup Manager are in qtdefines.py for any future use;
- Possibly redundant but still added: DlgWelcomeScreen.

PartySensei committed:
Setup Dialog feature

goatpig commented on ArmoryQt.py:
```diff
@@ -5236,15 +5248,21 @@ def unregisterProgressCallback(self, id):
+   #    Start cppbridge now and wire the callback (skip if already started during setup)
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if not getattr(TheBridge.bridgeSocket, 'run', False):
```
what is going on here??

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+MIN_CPU_CORES = 2
+
+################################################################################
+def createStyledLabel(text, alignment=QtCore.Qt.AlignLeft):
```
this is copy pasta Qt boilerplate from migration wizard, centralize that code into a single file

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+
+import os
+from qtpy import QtCore, QtGui, QtWidgets
+from qtdialogs.qtdefines import QRichLabel, apply_dialog_base_style, UI_STYLE_FRAME, UI_STYLE_BUTTON_STANDARD
```
80 char per lines please

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+REQUIREMENTS_FRAME_STYLE = UI_STYLE_FRAME
+
+################################################################################
+class DlgWelcomeScreen(QtWidgets.QDialog):
```
can't this be an ArmoryDialog child?

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+    def createIntroSection(self):
+        """Create the introduction text section."""
+        introText = createStyledLabel("""
+            <p style="color: %s; font-size: 10pt; margin: 0;">
```
missing `tr` encapsulation

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+        except Exception as e:
+            print(f"Error checking system requirements: {str(e)}")
+            # If we can't check requirements, assume they're sufficient
+            return self._getDefaultRequirements()
```
other way around, allow the app to run is reqs cant be check

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+        except Exception:
+            return True  # Assume sufficient if check fails
+
+    def _checkDiskSpaceWindows(self):
```
the system checks should be implement in armoryengiune/ArmoryUtils.py, if they're not there already

goatpig commented on qtdialogs/DlgWelcomeScreen.py:
```diff
@@ -0,0 +1,359 @@
+        """Check disk space on Windows using Windows API."""
+        import ctypes
+        free_bytes = ctypes.c_ulonglong(0)
+        ctypes.windll.kernel32.GetDiskFreeSpaceExW(
```
there ought to be something better than ctypes to get diskspace in py3

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -60,6 +62,237 @@ def wrapper(*args, **kwargs):
+   'tab_hover': htmlColor('SlightBkgdLight')
+}
+
+def apply_dialog_base_style(widget):
```
dont mix camel and snake case

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -209,10 +442,10 @@ def initialColResize(tblViewObj, sizeList):
-      hAlign=QtCore.Qt.AlignLeft,
-      vAlign=QtCore.Qt.AlignVCenter,
-      **kwargs):
+   def __init__(self, txt, doWrap=True, \
```
what is this abomination of an indent align? revert this

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -245,14 +478,14 @@ def setBold(self):
-      **kwargs):
-      super(QRichLabel_AutoToolTip, self).__init__(txt,
-         doWrap, hAlign, vAlign, **kwargs)
+   def __init__(self, txt, doWrap=True, \
```
why do you use `\` here but not in later declarations? make your style consistent

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -462,7 +697,8 @@ def restoreTableView(qtbl, hexBytes):
          qtbl.setColumnWidth(i, c)
    except Exception as e:
-      traceback.print_tb(e.__traceback__)
+      print('- Error loading table view -')
```
logging.error

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -625,35 +864,40 @@ def createBitmap(imgMtrx2D, writeToFile=-1, returnBinary=True):
    else:
-      fullPath = unicode(QtWidgets.QFileDialog.getExistingDirectory(par, title, initPath, \
-                                       options=QtWidgets.QFileDialog.DontUseNativeDialog))
+      fullPath = QtWidgets.QFileDialog.getExistingDirectory(par, title, initPath, \
```
clean up indent inbreeding and wild `\` where you find them too

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+         LOGINFO("Creating wallet tab...")
+         # Ensure the bridge is running so wallet listing works in this dialog
+         try:
+            if (TheBDM is not None and
```
TheBridge is a singleton, start it before anything else and then just please, stop trying to start it.

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+      try:
+         if self.bitcoinCoreProcess is not None:
+            # Send SIGTERM signal to gracefully shut down
+            self.bitcoinCoreProcess.terminate()
```
bridge will deal with this, do not replicate process automation in python

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+         mainLayout.setSpacing(UI_FRAME_PADDING)
+
+         # Directory Settings Frame
+         LOGINFO("Creating directory frame...")
```
get rid of these info entries

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+         self.p2pPortInput.setEnabled(True)
+         self.rpcPortInput.setEnabled(False)  # Keep RPC port disabled
+
+   def autoDetectCoreSettings(self):
```
get rid of this, seed paths from bridge where necessary

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+            self.tr('Failed to auto-detect settings: {}').format(str(e))
+         )
+
+   def checkBitcoinCoreRunning(self):
```
ask the bridge

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+      except Exception as e:
+         LOGEXCEPT("Error focusing Bitcoin Core window: %s", str(e))
+
+   def startBitcoinCoreNode(self):
```
bridge will handle this

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+      self.walletList.clear()
+      try:
+         # Ask bridge; if bridge not ready, show a placeholder and exit
+         if (TheBDM is None or
```
this is overriding your try block exception handling

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+         # List wallets using wallet manager (returns a List of WalletFileData)
+         wltList = TheBridge.wltManager.listWallets()
+         # Iterate by index to satisfy capnp dynamic list reader
+         for idx in range(len(wltList)):
```
`for entry in wltList:`
capnp lists are natively iteratable

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+         for idx in range(len(wltList)):
+            entry = wltList[idx]
+            # Derive display values
+            walletPath = getattr(entry, 'path', '')
```
dont abuse getattr

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+            # WalletManagerReply.WalletLoadState: 1=legacy, 3=encrypted, 4=ready
+            # loadState is a capnp enum; rely on underlying numeric value via string compare
+            loadStateStr = str(loadState)
+            isLegacy = (loadStateStr.endswith('.legacy')) or (not bool(walletId))
```
there is a straight up field for legacy status, use that instead of string manipulation

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+      try:
+         # DlgMigrateWallet(promptId, wltID, verbose, parent=None, main=None)
+         DlgMigrateWallet(None, walletPath, self.tr('Migrate'), self, self).exec_()
+         TheBridge.wltManager.loadWallets()
```
isnt this natively called by the migration wizard already?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+   def _unlockWallet(self, walletId):
+      try:
+         # Control header unlock requires a path; list again to find the entry
+         wltList = TheBridge.wltManager.listWallets()
```
dont ask BE for wallets when py side store has them. this will fuck a bunch of things when multiple wallets are present

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+   def _loadWallet(self, walletId):
+      try:
+         # Stage and refresh
+         TheBridge.wltManager.stageWallet(walletId, True)
```
why is this called `_loadWallet` and all it does is staging a wallet? Why are you calling `loadWallets` after flipping a single boolean? this stuff is meant to run on more than only a rtx5090

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+            )
+            return
+
+         # Attempt to connect to the remote database
```
ask cppbridge, dont blindly carry over old py code

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+         self.remoteHostEdit.setText(TheSettings.getSettingOrSetDefault('RemoteDBHost', ''))
+         self.remotePortEdit.setText(TheSettings.getSettingOrSetDefault('RemoteDBPort', ''))
+         self.remoteUserEdit.setText(TheSettings.getSettingOrSetDefault('RemoteDBUser', ''))
+         self.remotePasswordEdit.setText(TheSettings.getSettingOrSetDefault('RemoteDBPass', ''))
```
no, no, no

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+            TheSettings.set('RemoteDBHost', self.remoteHostEdit.text())
+            TheSettings.set('RemoteDBPort', self.remotePortEdit.text())
+            TheSettings.set('RemoteDBUser', self.remoteUserEdit.text())
+            TheSettings.set('RemoteDBPass', self.remotePasswordEdit.text())
```
wtf...

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1197 @@
+      self.cancelButton = None
+      
+      # Widget references for settings
+      self.satoshiHomePath = None
```
this is too hard to read.

break it down into relevant feature groups, implement the sanity checks there. It's fine to have a flat implementation for GUI, but the underling data should be isolated where possible.

PartySensei committed:
Fix lean wallet listing, callback-driven migrate/unlock, style cleanup
Lean wallet listing, Migration wiring, Unlock wiring, UI/UX and code
style, cleanup and safety, net effect

goatpig commented on ArmoryQt.py:
```diff
@@ -559,11 +560,13 @@ def exportTx():
       actCloseApp    = self.createAction(self.tr('&Quit Armory'), self.closeForReal)
       self.menusList[MENUS.File].addAction(actExportTx)
       self.menusList[MENUS.File].addAction(actSettings)
+      self.menusList[MENUS.File].addAction(actSetupManager)
```
what is the purpose of the setup manager after armory has started?

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+
+   #    Show Setup Manager (modal and focused)
+   #    Bridge is already running; dialog must not try to start it
+   from qtdialogs.DlgSetupManager import DlgSetupManager
```
already imported at the top of the file

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+   # 1) Start the bridge BEFORE anything else (singleton) so dialogs can query it
+   #    Start with a no-op callback; we'll wire readiness to the dialog instance
+   #    below to avoid overwriting global callbacks.
+   TheBDM.startBridge(getBridgeArgList(), lambda *args, **kwargs: None)
```
wtf is going on with this wild `lambda`?

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+
+   dlg = DlgSetupManager(parent=None, main=None)
+   try:
+      from armoryengine.CppBridge import TheBridge
```
already imported at the top of the file. stop using local imports, it makes no sense anyways, it all goes into a general import dict

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+            dlg, 'onBridgeReady', QtCore.Qt.QueuedConnection)
+      else:
+         TheBridge.bridgeSocket.bip15xConnection.setNotifyReadyLbd(
+            lambda: QtCore.QMetaObject.invokeMethod(
```
dont use invokeMethod or qt slot decorators, use `TheSignalExecution`. dont introduce new ways of doing stuff that is already available in the code base. if in doubt, ask

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+   dlg = DlgSetupManager(parent=None, main=None)
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if TheBridge.bridgeSocket.bip15xConnection.ready():
```
this whole block makes no sense:

1. why check if the bridge is already running? this is the entry point to the process, it 100% knows if the bridge `start` method has been called by now
2. you can pass a ready callback to `TheBDM.startBridge`, instead you do some lambda nonsense and access the inner connection object to force a callback after the start.
3. By refusing to use the code as its designed, you are introducing 2 race conditions: if the bridge readies up before you set the callback, it will never fire. If the dialog isn't started before the callback fires, it will try to progress over an undefined state of DlgSetupManager.

This is bad, use the code as it is meant to, don't adlib the start sequence.

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if TheBridge.bridgeSocket.bip15xConnection.ready():
+         QtCore.QMetaObject.invokeMethod(
```
what is going on here? why qt invoke from the thread that calls `exec_`?

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+
+   if dlg.exec_() != QtWidgets.QDialog.Accepted:
+      try:
+         from armoryengine.CppBridge import TheBridge
```
stop using local imports

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+         pass
+      sys.exit(1)
+
+   # 2) Splash screen appears
```
this is silly. the splash screen is there to show responsiveness to the user while the process starts. it should come first, while the bridge is loading up, not in between the bridge and the armoryqt, where it would be instant

goatpig commented on ArmoryQt.py:
```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
+         from armoryengine.CppBridge import TheBridge
+         TheBridge.service.shutdown()
+      except Exception:
+         pass
```
Dont handle exceptions silently. If you do not know what to do with the exception, then don't catch it at all

goatpig commented on ArmoryQt.py:
```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
+         if OS_WINDOWS and sys.frozen:
+            app_dir = os.path.dirname(sys.executable)
+      except Exception:
+         pass
```
silent exception handling again

goatpig commented on ArmoryQt.py:
```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
-   TheBDM.startBridge(getBridgeArgList(), armoryMainWindow.networkReadyCallback)
+   # Bridge already started; ensure the main window receives the ready callback
+   try:
+      from armoryengine.CppBridge import TheBridge
```
local imports...

goatpig commented on ArmoryQt.py:
```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
+   # Bridge already started; ensure the main window receives the ready callback
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if TheBridge.bridgeSocket.bip15xConnection.ready():
```
accessing the connection object in TheBridge again, checking on bridge ready again...

goatpig commented on ArmoryQt.py:
```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
+      else:
+         TheBridge.bridgeSocket.bip15xConnection.setNotifyReadyLbd(
+            lambda: QtCore.QMetaObject.invokeMethod(
+               armoryMainWindow, 'networkReadyCallback', QtCore.Qt.QueuedConnection))
```
how? how can you get this far without the bridge running and setup manager handling the ready callback?

goatpig commented on ArmoryQt.py:
```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
-   #start cppbridge
-   TheBDM.startBridge(getBridgeArgList(), armoryMainWindow.networkReadyCallback)
+   # Bridge already started; ensure the main window receives the ready callback
+   try:
```
what does this block do? spawn armoryqt from the setup manager on setup completion instead

goatpig commented on ArmoryQt.py:
```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
+      except Exception:
+         pass
    translator.load(TheSettings.getGuiLanguage(), os.path.join(app_dir, "lang/"))
    QAPP.installTranslator(translator)
```
so the setup manager shouldn't get any translations?

PartySensei committed:
Refactor ArmoryQt.py and DlgSetupManager.py
- Adhere to maintainer feedback
- Replace QtCore.QTimer.singleShot with TheSignalExecution.callLater
- Fix setup manager -> main window architectural flow via callback
  pattern
- Remove local imports and silent exception handling
- Fix splash screen timing and wallet list population on startup

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -363,11 +362,8 @@ def showEvent(self, event):
-            self.refreshButton.setEnabled(True)
-      except Exception:
-         pass
+      if self.refreshButton:
```
why is there an explicit refresh button?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+      self.remotePasswordEdit = None
+      self.testConnectionButton = None
+      self.walletIdToCheckbox = {}
+      self.bridgeReady = False
```
This isn't a useful flag. Callback driven flow should guarantee the user cant get to the GUI without the bridge running

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+         self.saveSettings()
+
+         # Mark setup as completed and close dialog
+         TheSettings.set('SetupManagerCompleted', True)
```
what's the point of setting this flag if it's never read?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+      """Handle dialog rejection without marking setup as completed."""
+      super(DlgSetupManager, self).reject()
+      # Don't set SetupManagerCompleted to True when rejected
+      TheSettings.set('SetupManagerCompleted', False)
```
again, this flag is never read, only ever set

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+
+   def closeEvent(self, event):
+      """Handle dialog close event."""
+      super().closeEvent(event)
```
is there a need to override this if all it does is call the parent?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+
+   def browseSatoshiHome(self):
+      """Open a directory dialog to select the Bitcoin Core data directory."""
+      dir = QtWidgets.QFileDialog.getExistingDirectory(
```
use something else than `dir`, it overwrites a reserved keyword according to the linter

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+
+   def browseDirDialog(self, lineEdit):
+      """Open directory chooser into the given QLineEdit using shared helper."""
+      from qtdialogs.qtdefines import selectDirectoryForQLineEdit
```
avoid local imports

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+   def networkModeChanged(self, index):
+      """Handle changes to the network mode selection."""
+      # Update P2P port based on network mode
+      if index == 0:  # Mainnet
```
you cant change network mode after the bridge has started

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+
+      browseBtn = QtWidgets.QPushButton(self.tr("Browse..."))
+      browseBtn.setFixedWidth(100)  # Match Accept/Cancel buttons
+      def browseArmoryDataDir():
```
line break

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+            self.walletList.setItemWidget(item, 0, cell)
+            self.walletIdToCheckbox[walletId] = cb
+
+         fullPath = walletPath if walletPath else os.path.join(self.armoryDataDirEdit.text(), fileName)
```
full path isnt necessary so long as the datadir is displayed somewhere on the tab

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+      try:
+         stateVal = entry.state
+         # Handle numeric enums and string fallbacks defensively
+         if isinstance(stateVal, int):
```
where are the numeric enums coming from?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+      """
+      try:
+         if self.main is not None:
+            walletData = TheBridge.utils.importWallet(walletPath)
```
i dont understand this at all. why call import when the file is listed in the datadir?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+            return
+
+         # Fallback: run migration directly via the bridge (no wizard UI)
+         import uuid
```
stop with the local imports, even more so on something as ubiquitous as `uuid`

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+   def migrateWallet(self, walletPath):
+      """Start migration for a legacy wallet.
+
+      - If a valid main window is available, use the full migration dialog.
```
this condition makes no sense. setupmanager is meant to condition the main window, why would it be spawned after the main window, ever?

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+      if success:
+         QtWidgets.QMessageBox.information(
+            self,
+            self.tr('Migration Complete'),
```
i dont think there's a need for an explicit message box on success, it can just update the state/type/action for the given wallet on completion, i.e. simply reloading the list

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+         originallyStaged = bool(rowMeta.get('staged')) if rowMeta else False
+         try:
+            if walletId and not originallyStaged:
+               TheBridge.wltManager.stageWallet(walletId, True)
```
this isn't a correct use the wallet manager API
read the cpp tests, it displays the correct flow:
https://github.com/goatpig/BitcoinArmory/blob/dev/cppForSwig/gtest/BridgeTests.cpp#L1565

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1268 @@
+         'remoteHost': self.remoteHostEdit.text() if dbScenario == 'Connect to Remote Database' else '',
+         'remotePort': self.remotePortEdit.text() if dbScenario == 'Connect to Remote Database' else '',
+         'remoteUser': self.remoteUserEdit.text() if dbScenario == 'Connect to Remote Database' else '',
+         'remotePass': self.remotePasswordEdit.text() if dbScenario == 'Connect to Remote Database' else '',
```
there are no passwords to manipulate, the bridge will prompt you with an unlock callback if you try to load a locked peers db, same as with wallets

PartySensei committed:
Fix maintainer feedback violations in DlgSetupManager.py
- Remove explicit refresh button and unused settings flags
- Remove password inputs
- Simplify migration flow
- Add wallet state enums and remove unecessary messaging
- Import and other code cleanup

PartySensei committed:
Fix list/migrate/stage/unlock flow and wallet list display

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -1,10 +1,12 @@
 # See LICENSE or http://www.gnu.org/licenses/agpl.html                       #
 #                                                                            #
-# Copyright (C) 2016-2025, goatpig                                           #
+# Copyright (C) 2016-2024, goatpig                                           #
```
what's wrong with 2025? changes happened this year, dont backdate it

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -264,19 +564,21 @@ def event(self, event):
 class QMoneyLabel(QRichLabel):
    def __init__(self, nSatoshi, ndec=8, maxZeros=2, wColor=True,
-      wBold=False, txtSize=10):
+                              wBold=False, txtSize=10):
```
bad indent

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -264,19 +564,21 @@ def event(self, event):
+
    def setValueText(self, nSatoshi, ndec=None, maxZeros=None, wColor=None,
-      wBold=None, txtSize=10):
+                                             wBold=None, txtSize=10):
```
bad indent

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -818,3 +1129,46 @@ def getKdfBytes(self):
          pass
       return kdfBytes
+
+LOGERROR = logging.error
```
import LOGERROR from armoryengine.ArmoryUtils, don't adlib it

goatpig commented on qtdialogs/qtdefines.py:
```diff
@@ -13,10 +15,13 @@
 from tempfile import mkstemp
 import urllib
-import traceback
+import logging
```
use the dedicated LOG wrappers from ArmoryUtils, don't import the native package

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1291 @@
+         # Core
+         TheSettings.set('SatoshiDatadir', paths['core'])
+         TheSettings.set('ManageSatoshi', core['manageSatoshi'])
+         TheSettings.set('NetworkMode', core['networkMode'])
```
network mode is set by cli args only

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1291 @@
+   def testRemoteConnection(self):
+      """Delegate connection setup to the bridge; no local management."""
+      try:
+         TheBridge.service.setupDB()
```
test connection isnt setupDB

goatpig commented on qtdialogs/DlgSetupManager.py:
```diff
@@ -0,0 +1,1291 @@
+
+         # Trigger wallet loading - bridge will prompt for unlock if needed
+         def loadWalletsAsync():
+            TheBridge.wltManager.loadWallets()
```
loadWallets != listWallets

PartySensei committed:
refactor: Address maintainer feedback in setup dialog components
- Fix network mode UI
- Remove setupDB() call from testRemoteConnection method
- Fix import organization and parameter defaults in qtdefines
- Remove silent exception handling patterns
- Clean up code formatting and style issues
